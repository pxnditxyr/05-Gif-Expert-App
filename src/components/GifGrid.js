// import React, { useState, useEffect } from 'react';
import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({ category }) => {


	const { data, loading } = useFetchGifs();
	console.log( data );
	console.log( loading );
	// const [ images, setImages ] = useState( [] );
	// useEffect( () => {
	// 	getGifs( category )
	// 		.then( imgs => {
	// 			setImages( imgs );
	// 		});
	// }, [ category ])


	return (
		<>
			<h2> { category } </h2>
			{ loading ? 'Loading...' : 'Data charge done' }
			{/* <div className="card-grid"> */}
			{/* 	{ */}
			{/* 		images.map( img => ( */}
			{/* 			<GifGridItem */}
			{/* 				key={ img.id } */}
			{/* 				{ ...img  } */}
			{/* 			/> */}
			{/* 		)) */}
			{/* 	} */}
			{/* </div> */}
		</>
	);
};

GifGrid.prototype = {
	category: PropTypes.string.isRequired,
}
GifGrid.defaultProps = {
	category: "Nuevo"
}
