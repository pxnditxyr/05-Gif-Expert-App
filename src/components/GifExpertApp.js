import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

	// const categories = [ "Naruto", "Bleach", "One Piece" ];
	//const [ categories, setCategories ] = useState( ["Naruto", "Bleach", "One Piece", "KissxSis"] );
	const [ categories, setCategories ] = useState( ["Naruto"] );


	// const handleAdd = () => setCategories( [...categories, "KissxSis"] )

	return (

		<>
			<h2> GifExpertApp XD </h2>
			<AddCategory setCategories={ setCategories } />
			<hr />
			{/* <button onClick={ handleAdd }> Add </button> */}
			<ol>
				{
					categories.map( category =>
						<GifGrid
							key={ category }
							category={ category }
						/>
						//{/* <li key={ category }> { category } </li> */}
					)
				}
			</ol>
		</>
	);
};
