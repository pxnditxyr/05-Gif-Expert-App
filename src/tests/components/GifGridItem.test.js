import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe( "Tests on GifGridItem component", () => {

	const title = 'Hello my love';
	const url = 'http://localhost/something.jpg';
	const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

	test( "Must be show the component", () => {
		expect( wrapper ).toMatchSnapshot();
	});

	test( "Should have a title on a h3", () => {
		const h3 = wrapper.find( 'h3' );
		expect( h3.text().trim() ).toBe( title );
	});
	
	test( "Should have image equal to url and alt of props", () => {
		const img = wrapper.find( 'img' );
		expect( img.prop( 'src' ) ).toBe( url );
		expect( img.prop( 'alt' ) ).toBe( title );
	});

	test( "Should have animate__fadeIn", () => {
		const div = wrapper.find( 'div' );
		const classes = div.prop( 'className' ).split( " " );
		expect( classes.includes( "animate__fadeIn" ) ).toBe( true );
	});
});
