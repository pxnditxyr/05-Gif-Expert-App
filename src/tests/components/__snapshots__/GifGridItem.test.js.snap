// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItem component Must be show the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hello my love"
    src="http://localhost/something.jpg"
  />
  <h3>
     
    Hello my love
     
  </h3>
</div>
`;
